### PR TITLE
NXP-24309: change supported zip mime types in zip to html converter.

### PR DESCRIPTION
**Pull request by richardsd**

### CODE REVIEW

File: nuxeo-services/nuxeo-platform-convert/src/test/java/org/nuxeo/ecm/platform/convert/tests/TestZip2HtmlConverter.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package org.nuxeo.ecm.platform.convert.tests;
```
Comment (efge) [inline nuxeo-services/nuxeo-platform-convert/src/test/java/org/nuxeo/ecm/platform/convert/tests/TestZip2HtmlConverter.java]: Please add the standard copyright header on this file.

File: nuxeo-services/nuxeo-platform-convert/src/test/java/org/nuxeo/ecm/platform/convert/tests/TestZip2HtmlConverter.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class TestZip2HtmlConverter extends BaseConverterTest{
```
Comment (efge) [inline nuxeo-services/nuxeo-platform-convert/src/test/java/org/nuxeo/ecm/platform/convert/tests/TestZip2HtmlConverter.java]: Space before `{`

File: nuxeo-services/nuxeo-platform-convert/src/test/java/org/nuxeo/ecm/platform/convert/tests/TestZip2HtmlConverter.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+        BlobHolder result = cs.convert(converterName, htmlBH, parameters);
+        assertNotNull(result);
+
+        List<Blob> blobs = result.getBlobs();
+        assertNotNull(blobs);
+        assertEquals(3, blobs.size());
+
+        String content = blobs.get(0).getString();
+        assertTrue(content.contains("Hello") || content.contains("hello"));
```
Comment (efge) [inline nuxeo-services/nuxeo-platform-convert/src/test/java/org/nuxeo/ecm/platform/convert/tests/TestZip2HtmlConverter.java]: Can you do a more precise check? Comparing the string with exactly what you expect? You should check a specific blob as well, not just the first one (not sure you'll have a deterministic order).

File: nuxeo-services/nuxeo-platform-convert/src/test/java/org/nuxeo/ecm/platform/convert/tests/TestZip2HtmlConverter.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    public void testConvertToMimeType(String mimeType) throws IOException {
+        String converterName = cs.getConverterName(mimeType, "text/html");
+        assertEquals("zip2html", converterName);
+
+        checkConverterAvailability(converterName);
+
+        BlobHolder htmlBH = getBlobFromPath("test-docs/hello.zip");
+        htmlBH.getBlob().setMimeType(mimeType);
+        Map<String, Serializable> parameters = new HashMap<>();
```
Comment (ataillefer) [inline nuxeo-services/nuxeo-platform-convert/src/test/java/org/nuxeo/ecm/platform/convert/tests/TestZip2HtmlConverter.java]: Could use `Collections.emptyMap()` instead.

**Comment by ataillefer** [comment]
:+1:

File: nuxeo-services/nuxeo-platform-convert/src/test/java/org/nuxeo/ecm/platform/convert/tests/TestZip2HtmlConverter.java (hunk 3 above)
Comment (richardsd) [reply nuxeo-services/nuxeo-platform-convert/src/test/java/org/nuxeo/ecm/platform/convert/tests/TestZip2HtmlConverter.java]: Yes, you are correct. I do think the order is deterministic. I'll improve it.

**Comment by efge** [comment]
Please comment on the PR after you re-push modified code, otherwise nobody is notified that there is something new to review.

File: nuxeo-services/nuxeo-platform-convert/src/test/java/org/nuxeo/ecm/platform/convert/tests/TestZip2HtmlConverter.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        BlobHolder result = cs.convert(converterName, htmlBH, parameters);
+        assertNotNull(result);
+
+        List<Blob> blobs = result.getBlobs();
+        assertNotNull(blobs);
+        assertEquals(3, blobs.size());
+
+        String content = blobs.get(0).getString();
+        System.out.println("The blob content");
+        System.out.println(content);
```
Comment (efge) [inline nuxeo-services/nuxeo-platform-convert/src/test/java/org/nuxeo/ecm/platform/convert/tests/TestZip2HtmlConverter.java]: Remove `System.out.println`

Comment (richardsd) [reply nuxeo-services/nuxeo-platform-convert/src/test/java/org/nuxeo/ecm/platform/convert/tests/TestZip2HtmlConverter.java]: Florent I was just testing something before notifying, forgot the print there. I will update the PR once the T&P is finished 👍

**Comment by efge** [comment]
👍

**Comment by richardsd** [comment]
https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-rdias-master/47/

**Pull request merged**